dapps/v2Explorer/src/utils/UniversalProvider.tsx

```diff
@@ -46,11 +53,29 @@ export async function createUniversalProvider() {
     });
 
     // Subscribe to session delete
-    universalProvider.on('session_delete', ({id, topic}) => {
-      console.log('session_delete', id, topic);
-    });
+    universalProvider.on(
+      'session_delete',
+      ({id, topic}: {id: string; topic: string}) => {
+        onSessionDisconnect?.({id, topic});
+        console.log('session_delete', id, topic);
+      },
+    );
+  } catch {
+    console.log('Error for connecting');
```

Comment:
Web version of web3modal has a small notification for errors, would be cool to show something similar in error cases instead of logs, perhaps remove log here and add another todo?

Reply:
should we use our own UI or add some callbacks and let the apps handle the errors?

Reply:
We do both on web i.e. 

hooks
https://docs.walletconnect.com/2.0/web3modal/react/hooks#useweb3modal

components
https://docs.walletconnect.com/2.0/web3modal/react/components#web3button

Reply:
For now, we can show our own small notification ui, but there will also be a callback mechanism in web version of web3modal like you mentioned.

You could add something in now or wait till it's added on web one to align with the api.

Reply:
ok! talked with tom and decided to use native alerts for now